052-connector-state.md

```diff
@@ -0,0 +1,77 @@
+<!-- This template is provided as an example with sections you may wish to comment on with respect to your proposal. Add or remove sections as required to best articulate the proposal. -->
+
+# Support stopping connectors
+
+This proposal aims at adding support for stopping Kafka Connect connectors.
+
+## Current situation
+
+At the moment Strimzi allows pausing connectors. This is done by a property, `pause`, in the `KafkaConnectorSpec` schema. Whenever it is set to `true`, the connectors and its tasks are paused.
+
+For example:
+```yaml
+spec:
+  class: org.apache.kafka.connect.file.FileStreamSourceConnector
+  config:
+    file: /opt/kafka/LICENSE
+    topic: my-topic
+  pause: true
+  tasksMax: 1
+```
+
+## Motivation
+
+Since Kafka 3.5, Kafka Connect connectors can also be stopped. This is done by using a new REST endpoint `PUT /connectors/{connector}/stop`. Compared to paused where the connector and tasks stay instantiated, when stopped only the configuration of the connector is kept but nothing is actually running.
```

Comment:
How will it be handled in Kafka 3.4?

Reply:
Kafka 3.4 does not have this endpoint so connectors can't be stopped. Will the next Strimzi version still support Kafka < 3.5.0?

Reply:
In general, we support two last minor Kafka versions. So once we add 3.6 support, we will likely drop 3.4 support. (I guess depending on Kafka schedules, but likely sometime shortly before the release of Strimzi 0.38)

I guess waiting is one option. Other option is simply documenting it as a known issue and handling it in the code so that it does not error.

Reply:
Is it possible to prevent setting `state` to `stop` is the Kafka version is < 3.5? For example like what happens if you try setting `pause` to a value different than `true` or `false`?

Reply:
Not really. The CRD exists only once. So you can prevent it from being set to `stop` -> but for all versions only.

Reply:
I pushed an update to mention that if this is merged before support for Kafka 3.4 is dropped, the docs will mention it as a known issue.